torchtune/models/flamingo/_encoders.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+from torch import nn
+
+from torchtune.modules import (
+    Fp32LayerNorm,
+    GroupedQueryAttention,
+    TanhGate,
+    TransformerSelfAttentionLayer,
+)
+
+
+class FlamingoVisionAdapter(nn.Module):
```

Comment:
I don't think this follows the design pattern of our other components. Why can't we just pass layer and projection as nn.Modules to the init so that we keep things generic? Then use builders to actually construct the lower-level components and piece them all together